### ISSUE TITLE
💬 RFC:  Ideas for improving the Error State pages

### ISSUE THREAD

**Issue by ShiranHi** [backstage/backstage#23704]
### 🔖 Need

Currently, the error state displays an overwhelming amount of textual information and links, making it challenging to discern the specific error type. Clearly identifying the type of error aids users in resolving issues, and creating a standardized solution that addresses different error types can prove beneficial.

### 🎉 Proposal

I propose updating the current title to clearly indicate the error number along with a brief description. For instance, "404 Page not found" or "500 Internal Server Error". Additionally, expanding the displayed information with a "Read more" button to reveal further details after two lines can prevent users from feeling overwhelmed by content that may not necessarily be read. Another enhancement could involve including only two buttons to guide users towards their next actions. 
Introducing a custom image may also improve the page's visual appeal, and centering all content could assist users in quickly locating essential information.

Please find below a design suggestion, I'd appreciate your feedback on it

<img width="3160" alt="Error state" src="https://github.com/backstage/backstage/assets/77277517/2fe342da-404d-4c55-9993-6be447017a5f">

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by awanlin** [comment]
Just a note that the image you have as current is not the current page. This should get you that: https://demo.backstage.io/fake/url. Just want there to be a fair comparison. Also, there was recent changes around the error status code and showing the actual exception to help with getting more details.

You suggestion seem pretty good even with the above 👍

**Comment by ShiranHi** [comment]
thank you @awanlin for the feedback.

Here is the design with the same image:
<img width="3160" alt="Error state" src="https://github.com/backstage/backstage/assets/77277517/f3630f86-88cd-4ee4-9555-64cc9e27eabd">

**Comment by awanlin** [comment]
Hmm, that's still not right, I've never seen that "try these pages instead" part before. You sure that's not something custom you added? 🤔

**Comment by ShiranHi** [comment]
@awanlin I took this screen from this Figma file: https://backstage.io/docs/dls/figma/

**Comment by ShiranHi** [comment]
> I probably wouldn't use that as it's not representative of the real world. I'm not sure how old that Figma is but feels very out of date. I would probably spin up a Backstage instance to make proper comparisons.

you're right, I took a screenshot from https://demo.backstage.io/create page

![image](https://github.com/backstage/backstage/assets/77277517/3fea09ff-524b-419a-85d6-9c1211c1fc88)

I updated the image in the first comment

**Comment by Rugvip** [comment]
I like the proposed design! 👍 The blue title might be a bit tricky just because we might not have a good token for that in our palette, but perhaps keeping it the default color is alright? Maybe a detail to handle in an eventual PR.

Gonna be looking for help through contributions to implement this one 🙏

**Comment by awanlin** [comment]
Just wanted to share this PR around being able to provide a totally custom error page: https://github.com/backstage/backstage/pull/23532

I still think the default one would be much better with the suggestions being made here though. 

@ShiranHi, we recently added a feature to show the full stack trace on this page, where would that fit in this design? I'm not sure if having it under the "Read more" section would make it obvious that these details were available but I'm not personally too tied to that. Would like to hear your ideas though?

**Comment by ShiranHi** [comment]
Thank you for sharing @awanlin.
This is a good question, I proposed showing the start of the error message to avoid info overload. I think it could fit in a "Read more" section. Another approach will be adding a button for the full stack trace that will be shown in a modal, if it's long. What do you think?

**Comment by vinzscam** [comment]
I think help wanted to start discussing the proposed changes in a PR! 😄

**Comment by ShiranHi** [comment]
@awanlin I found out that we can only customize the 404 error page. I wanted to confirm if there are any other error pages that can be aligned with the same UI (such as 500, 502, 503...)

**Comment by awanlin** [comment]
Interesting, what have you tried @ShiranHi?

**Comment by ShiranHi** [comment]
@awanlin I was considering using a consistent layout for all error pages. The title will display the error number, and we'll place a custom image on the right side.

<img width="4630" alt="Error states" src="https://github.com/user-attachments/assets/a4c5f0ad-3426-4553-94a2-c2572887026e">

**Comment by logonoff** [comment]
Hey! I'd like to pick up this RFC if it's still open :)
